File: hospital.client/src/types/MedicineInventoryResponse.ts
import type { BranchResponse } from "./BranchResponse";
import type { MedicineResponse } from "./MedicineResponse";

export interface MedicineInventoryResponse {
  id: number;
  medicineId: number;
  branchId: number;
  currentStock: number;
  rowVersion?: number | null;
  state: number;
  createdAt: string;
  createdBy: number;
  updatedBy?: number | null;
  updatedAt?: string | null;

  medicine?: MedicineResponse | null;
  branch?: BranchResponse | null;
}

export interface MedicineInventoryRequest {
  id?: number | null;
  medicineId?: number | null;
  branchId?: number | null;
  currentStock?: number | null;
  state?: number | null;
  createdBy?: number | null;
  updatedBy?: number | null;
}

File: Hospital.Server/Entities/Response/MedicineInventoryResponse.cs
namespace Hospital.Server.Entities.Response
{
    public class MedicineInventoryResponse
    {
        public long Id { get; set; }
        public long MedicineId { get; set; }
        public long BranchId { get; set; }
        public int CurrentStock { get; set; }
        public uint RowVersion { get; set; }
        public int State { get; set; }
        public string CreatedAt { get; set; } = string.Empty;
        public long CreatedBy { get; set; }
        public long? UpdatedBy { get; set; }
        public string? UpdatedAt { get; set; }
    }
}
